File: services/ms-auth/src/main/java/airportsystem/msauth/repository/UsuarioRepository.java
package airportsystem.msauth.repository;

import airportsystem.msauth.model.Usuario;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UsuarioRepository extends MongoRepository<Usuario, String> {

    boolean existsByLogin(String login);
    boolean existsByCodigo(String codigo);

    Usuario findByLogin(String login);
}

File: services/ms-auth/airport-system.ms-auth/src/main/kotlin/com/tads/airport_system/ms_auth/repository/UsuarioRepository.kt
package com.tads.airport_system.ms_auth.repository

import com.tads.airport_system.ms_auth.model.Usuario
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface UsuarioRepository : MongoRepository<Usuario, String> {
    fun findByEmail(email: String): Usuario?
    fun existsByEmail(email: String): Boolean
} 